core/templates/services/math-interactions.service.ts

```diff
@@ -107,16 +109,31 @@ export class MathInteractionsService {
       return false;
     }
     try {
+      expressionString = this.insertMultiplicationSigns(expressionString);
       expressionObject = nerdamer(expressionString);
     } catch (err) {
       this.warningText = this.cleanErrorMessage(err.message, expressionString);
       return false;
     }
-    if (algebraic && expressionObject.variables().length === 0) {
-      this.warningText = 'It looks like you have entered only ' +
+    if (algebraic) {
+      let variablesList = expressionObject.variables();
+      if (variablesList.length === 0) {
+        this.warningText = 'It looks like you have entered only ' +
         'numbers. Make sure to include the necessary variables' +
         ' mentioned in the question.';
-      return false;
+        return false;
+      } else if (validVariablesList !== undefined &&
```

Comment:
Why would this be undefined? If it is for NumericExpression, make sure to manually pass empty array, don't condition for undefined.

Reply:
I've fixed this now.